sys/kern/vm_map.c

```diff
@@ -196,8 +196,22 @@ void vm_map_delete(vm_map_t *map) {
   pool_free(P_VM_MAP, map);
 }
 
-/* TODO: not implemented */
+/* XXX: it allows only to change protection of existing vm_map_entry */
 void vm_map_protect(vm_map_t *map, vaddr_t start, vaddr_t end, vm_prot_t prot) {
+  SCOPED_MTX_LOCK(&map->mtx);
+
+  vm_map_entry_t *ent;
+
+  klog("vm_map_protect: 0x%x - 0x%x %c%c%c", start, end,
+       (prot & VM_PROT_READ) ? 'r' : '-', (prot & VM_PROT_WRITE) ? 'w' : '-',
+       (prot & VM_PROT_EXEC) ? 'x' : '-');
+
+  ent = vm_map_find_entry(map, start);
+  assert(ent);
+  assert(ent->end == end);
+
+  ent->prot = prot;
+  pmap_protect(map->pmap, start, end, prot);
```

Comment:
IIUC it find's only first `vm_map_entry` - what with others? What is a semantic of this function?

Reply:
Okay, I have updated it. Function now can be used only with whole entries which are inside [start, end] region. I don't want to support splitting now because it will be used when `mprotect` is implemented. (Then I will also add detailed tests for that splitting.)